src/h5web/providers/mock/utils.ts

```diff
@@ -1,10 +1,34 @@
-import { mockMetadata } from './data';
-import { assertSimpleShape } from '../utils';
-import { assertDefined } from '../../visualizations/shared/utils';
-
-export function getMockDatasetDims(name: string): number[] {
-  const dataset = mockMetadata.datasets?.[name];
-  assertDefined(dataset, "Dataset doesn't exist");
-  assertSimpleShape(dataset);
-  return dataset.shape.dims;
+import { mockValues, mockMetadata } from './data';
+import {
+  assertDataset,
+  assertMySimpleShape,
+  assertNumericType,
+  isGroup,
+} from '../utils';
+import { assertArray, assertDefined } from '../../visualizations/shared/utils';
+import ndarray from 'ndarray';
+import { MyHDF5Entity } from '../models';
+import { getChildEntity } from '../../visualizations/nexus/utils';
+
+export function getMockDataArray(path: string): ndarray {
+  const pathSegments = path.slice(1).split('/');
+
+  const dataset = pathSegments.reduce<MyHDF5Entity | undefined>(
+    (parentEntity, currSegment) => {
+      return parentEntity && isGroup(parentEntity)
+        ? getChildEntity(parentEntity, currSegment)
+        : undefined;
+    },
+    mockMetadata
+  );
```

Comment:
This is the function we discussed, the one that is used in the stories. Here, the algorithm to find an entity from a path is taken from `useNxData`. In a future PR, I'll abstract this into a utility function, which I was thinking could even be thje existing `getEntityAtPath` (which currently accepts an array of indices).

Reply:
I would rename `path` to `absolutePath` to make clear it should begin by `/`